File: argentum-api/src/main/java/net/devium/argentum/rest/model/request/CoatCheckTagsRequest.java
package net.devium.argentum.rest.model.request;

import net.devium.argentum.jpa.CoatCheckTagEntity;
import net.devium.argentum.jpa.GuestEntity;

import java.math.BigDecimal;
import java.util.Date;
import java.util.List;
import java.util.stream.Collectors;

public class CoatCheckTagsRequest {
    private List<Long> ids;
    private Long guestId;
    private BigDecimal price;

    public List<Long> getIds() {
        return ids;
    }

    public Long getGuestId() {
        return guestId;
    }

    public BigDecimal getPrice() {
        return price;
    }

    public List<CoatCheckTagEntity> toEntities(GuestEntity guest) {
        Date time = new Date();
        return ids.stream()
            .map(id -> new CoatCheckTagEntity(id, time, guest))
            .collect(Collectors.toList());
    }
}

File: argentum-ui/src/app/common/rest-service/request/coat-check-tags-request.ts
import { Guest } from '../../model/guest';

export class CoatCheckTagsRequest {
  ids: number[];
  guestId: number;
  price: number;
}

export function fromCoatCheckTagIds(ids: number[], guest: Guest, price: number) {
  return {
    ids: ids,
    guestId: guest.id,
    price: price
  };
}
